From 5d296d3d3263c188b22038c699b7fdc159a27b7d Mon Sep 17 00:00:00 2001
From: Volodymyr Babchuk <volodymyr_babchuk@epam.com>
Date: Sat, 5 Feb 2022 02:19:43 +0200
Subject: [PATCH] plat_rcar_s4:  describe non-secure DDR to optee

OP-TEE needs to know about non-secure DDR layout in order to enabled
dynamic shared memory feature. This feature is crucial for
virtualization support in OP-TEE.

As currently ARM-TF can't provide DTB with memory layout - configure
it statically.

Signed-off-by: Volodymyr Babchuk <volodymyr_babchuk@epam.com>
---
 core/arch/arm/plat-rcar_gen4/conf.mk          |  2 +-
 core/arch/arm/plat-rcar_gen4/main.c           | 12 ------
 .../arch/arm/plat-rcar_gen4/platform_config.h | 41 ++-----------------
 3 files changed, 4 insertions(+), 51 deletions(-)

diff --git a/core/arch/arm/plat-rcar_gen4/conf.mk b/core/arch/arm/plat-rcar_gen4/conf.mk
index 56531a0e..196b3124 100644
--- a/core/arch/arm/plat-rcar_gen4/conf.mk
+++ b/core/arch/arm/plat-rcar_gen4/conf.mk
@@ -1,4 +1,4 @@
-PLATFORM_FLAVOR ?= generic_dt
+PLATFORM_FLAVOR ?= spider
 
 include core/arch/arm/cpu/cortex-armv8-0.mk
 
diff --git a/core/arch/arm/plat-rcar_gen4/main.c b/core/arch/arm/plat-rcar_gen4/main.c
index d4d48551..27481424 100644
--- a/core/arch/arm/plat-rcar_gen4/main.c
+++ b/core/arch/arm/plat-rcar_gen4/main.c
@@ -71,20 +71,8 @@ register_phys_mem_pgdir(DEVICE1_TYPE, DEVICE1_PA_BASE, DEVICE1_SIZE);
 register_phys_mem_pgdir(DEVICE2_TYPE, DEVICE2_PA_BASE, DEVICE2_SIZE);
 #endif
 
-/* Legacy platforms */
-#if defined(PLATFORM_FLAVOR_salvator_h3) || \
-	defined(PLATFORM_FLAVOR_salvator_h3_4x2g) || \
-	defined(PLATFORM_FLAVOR_salvator_m3) || \
-	defined(PLATFORM_FLAVOR_salvator_m3_2x4g)
 register_ddr(NSEC_DDR_0_BASE, NSEC_DDR_0_SIZE);
 register_ddr(NSEC_DDR_1_BASE, NSEC_DDR_1_SIZE);
-#ifdef NSEC_DDR_2_BASE
-register_ddr(NSEC_DDR_2_BASE, NSEC_DDR_2_SIZE);
-#endif
-#ifdef NSEC_DDR_3_BASE
-register_ddr(NSEC_DDR_3_BASE, NSEC_DDR_3_SIZE);
-#endif
-#endif
 
 void console_init(void)
 {
diff --git a/core/arch/arm/plat-rcar_gen4/platform_config.h b/core/arch/arm/plat-rcar_gen4/platform_config.h
index 76da2172..e65814ff 100644
--- a/core/arch/arm/plat-rcar_gen4/platform_config.h
+++ b/core/arch/arm/plat-rcar_gen4/platform_config.h
@@ -84,45 +84,10 @@
 
 #define MEM_SECTION_SIZE	(0x00100000U)
 
-#if defined(PLATFORM_FLAVOR_salvator_h3)
-#define NSEC_DDR_0_BASE		0x47E00000
-#define NSEC_DDR_0_SIZE		0x38200000
-#define NSEC_DDR_1_BASE		0x500000000U
-#define NSEC_DDR_1_SIZE		0x40000000
-#define NSEC_DDR_2_BASE		0x600000000U
-#define NSEC_DDR_2_SIZE		0x40000000
-#define NSEC_DDR_3_BASE		0x700000000U
-#define NSEC_DDR_3_SIZE		0x40000000
-
-#elif defined(PLATFORM_FLAVOR_salvator_h3_4x2g)
-#define NSEC_DDR_0_BASE		0x47E00000
-#define NSEC_DDR_0_SIZE		0x78200000
-#define NSEC_DDR_1_BASE		0x500000000U
+#define NSEC_DDR_0_BASE		0x48000000
+#define NSEC_DDR_0_SIZE		0x78000000
+#define NSEC_DDR_1_BASE		0x480000000U
 #define NSEC_DDR_1_SIZE		0x80000000U
-#define NSEC_DDR_2_BASE		0x600000000U
-#define NSEC_DDR_2_SIZE		0x80000000U
-#define NSEC_DDR_3_BASE		0x700000000U
-#define NSEC_DDR_3_SIZE		0x80000000U
-
-#elif defined(PLATFORM_FLAVOR_salvator_m3)
-#define NSEC_DDR_0_BASE		0x47E00000
-#define NSEC_DDR_0_SIZE		0x78200000
-#define NSEC_DDR_1_BASE		0x600000000U
-#define NSEC_DDR_1_SIZE		0x80000000U
-
-#elif defined(PLATFORM_FLAVOR_salvator_m3_2x4g)
-#define NSEC_DDR_0_BASE		0x47E00000
-#define NSEC_DDR_0_SIZE		0x78200000
-#define NSEC_DDR_1_BASE		0x400000000U
-#define NSEC_DDR_1_SIZE		0x80000000U
-#define NSEC_DDR_2_BASE		0x600000000U
-#define NSEC_DDR_2_SIZE		0x100000000U
-
-#else
-
-/* Generic DT-based platform */
-
-#endif
 
 #define DEVICE1_PA_BASE		ROUNDDOWN(CE_BASE, CORE_MMU_PGDIR_SIZE)
 #define DEVICE1_SIZE		(MEM_SECTION_SIZE * 2)
-- 
2.35.1

